File: Codescene.VSExtension.VS2022/Codescene.VSExtension.CodeSmells/Issues/Javascript/BumpyRoadExample.js
const fs = require('fs');
const path = require('path');

class BumpyRoadExample {
    processDirectory(dirPath) {

        const files = [];

        const items = fs.readdirSync(dirPath);

        items.forEach(item => {
            const fullPath = path.join(dirPath, item);

            if (fs.statSync(fullPath).isFile()) {
                if (/^data\d+\.csv$/.test(item)) {
                    files.push(fullPath);
                }
            }
        });

        let sb = "";

        files.forEach(filePath => {
            const data = fs.readFileSync(filePath, 'utf8');

            const lines = data.split(/\r?\n/);
            lines.forEach(line => {
                sb += line;
            });
        });

        fs.writeFileSync('data.csv', sb, 'utf8');
    }
}

const example = new BumpyRoadExample();
example.processDirectory('./some_directory');

File: Codescene.VSExtension.VS2022/Codescene.VSExtension.CodeSmells/Issues/CSharp/BumpyRoadExample.cs
using System.Collections.Generic;
using System.IO;
using System.Text;
using System.Text.RegularExpressions;

namespace Codescene.VSExtension.CodeSmells.Issues.CSharp
{
    class BumpyRoadExample
    {
        public void ProcessDirectory(string path)
        {
            var files = new List<string>();
            var directory = new DirectoryInfo(path);

            foreach (FileInfo fileInfo in directory.GetFiles())
            {
                if (Regex.IsMatch(fileInfo.Name, @"^data\d+\.csv$"))
                {
                    files.Add(fileInfo.FullName);
                }
            }

            var sb = new StringBuilder();
            foreach (string filePath in files)
            {
                using (var reader = new StreamReader(filePath))
                {
                    string line;
                    while ((line = reader.ReadLine()) != null)
                    {
                        sb.Append(line);
                    }
                }
            }

            using (var writer = new StreamWriter("data.csv"))
            {
                writer.Write(sb.ToString());
            }
        }
    }
}
